### PR TITLE
HARP-6117: Prebuild script for generating night&day mode Maki icons

### PR DESCRIPTION
**Pull request by kkostecki**
Script allows to generate texture atlases for night and day mode themes
using Maki icons set directly from Maki repository.
Internet connection is required to download assets.

Signed-off-by: Krystian Kostecki <ext-krystian.kostecki@here.com>

### CODE REVIEW

**Comment by kkostecki** [comment]
Assets required for icons creation are copied (via prepack.ts script) during yarn process. We may also commit them to this package directly.
Feel free to suggest some better names for scripts, yarn run commands etc...

File: @here/harp-map-theme/scripts/prepack.ts [hunk 1]
```diff
@@ -21,3 +21,7 @@ function onCopyError(err: Error) {
 const fontDir = path.dirname(require.resolve("@here/harp-font-resources/package.json"));
 
 ncp(path.join(fontDir, "resources"), path.join(__dirname, "..", "resources", "fonts"), onCopyError);
+
```
Comment (zbigg) [inline @here/harp-map-theme/scripts/prepack.ts]: Why this change is needed these ? 

Shouldn't all configs files like `maki-day-11.json` be here in this package ?

Comment (kkostecki) [reply @here/harp-map-theme/scripts/prepack.ts]: We may do it like that, I just saw that we are coping font resources from already existing package so done similar way (out harp-atlas-tools already have configs as examples). What do you suggest commit those configs to this package? If so, please tell me resources-dev/ is it correct path, or should I create some other folder?

Comment (zbigg) [reply @here/harp-map-theme/scripts/prepack.ts]: > we are coping font resources from already existing 

Yes, but this is only because font-resources are large and no-one wants to rebuild them on each built.

> out harp-atlas-tools already have configs as examples)

But they are examples, guys modyfying atlas tools and playing with examples shouldn't affect configs and assets for this particular use case - maki-based icons for `berlin` themes ...

Comment (kkostecki) [reply @here/harp-map-theme/scripts/prepack.ts]: Ok, fine for me too.

Comment (kkostecki) [reply @here/harp-map-theme/scripts/prepack.ts]: Done.

**Comment by kkostecki** [comment]
@heremaps-bot ok to test

File: @here/harp-map-theme/scripts/prepareIcons.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+async function generateMakiAtlas(
+    makiDir: string,
+    atlasPath: string,
+    smallIconsConfig: string,
+    bigIconsConfig: string
+): Promise<any> {
+    const cpus: number = os.cpus() ? os.cpus().length : 4;
+    // Create temprorary directory for sprites output inside makiDir
+    const tmpDir = tmp.dirSync({ template: "sprites-XXXXXX", dir: makiDir });
```
Comment (okanishchev) [inline @here/harp-map-theme/scripts/prepareIcons.ts]: tmpDir is not removed afterwards

File: @here/harp-map-theme/package.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "scripts": {
         "test": "cross-env mocha --require source-map-support/register $EXTRA_MOCHA_ARGS ./test/*.js",
         "build": "tsc",
+        "prepareIcons": "ts-node ./scripts/prepareIcons.ts",
```
Comment (okanishchev) [inline @here/harp-map-theme/package.json]: We have mostly dashed-style script names all over the project.
```suggestion
        "prepare-icons": "ts-node ./scripts/prepareIcons.ts",
```

Comment (kkostecki) [reply @here/harp-map-theme/package.json]: OK, will change...

**Comment by okanishchev** [comment]
I also think we should add all generated files to .gitignore

File: @here/harp-map-theme/scripts/prepareIcons.ts (hunk 2 above)
Comment (kkostecki) [reply @here/harp-map-theme/scripts/prepareIcons.ts]: It is, by higher level function which creates root folder for processing, see below:
```
const tmpDir = tmp.dirSync({ template: "resources-tmp-XXXXXX", dir: ".", unsafeCleanup: true });
...

 // Manual cleanup
tmpDir.removeCallback();
```
But maybe I could make it more explicit somehow.

Comment (okanishchev) [reply @here/harp-map-theme/scripts/prepareIcons.ts]: It's whether ```dir``` option to tmp.dirSync is not used, or some other reason, but these dirs are created in project root.

Comment (kkostecki) [reply @here/harp-map-theme/scripts/prepareIcons.ts]: I have just checked it again on my machine (Ubuntu 16.04). Folders created are logged during processing:
```
Input path: 'resources-tmp-eEJSFC/icons/*-11.svg' 
Output dir: 'resources-tmp-eEJSFC/sprites-6T5D7h' 
Input path: 'resources-tmp-eEJSFC/icons/*-15.svg' 
Output dir: 'resources-tmp-eEJSFC/sprites-6T5D7h' 
Input path: 'resources-tmp-eEJSFC/sprites-6T5D7h/*.png' 
Output path: 'resources/maki_icons_day[.png|.json]' 
Output dir: 'resources' 
Input path: 'resources-tmp-eEJSFC/icons/*-11.svg' 
Output dir: 'resources-tmp-eEJSFC/sprites-JgnfL5' 
Input path: 'resources-tmp-eEJSFC/icons/*-15.svg' 
Output dir: 'resources-tmp-eEJSFC/sprites-JgnfL5' 
Input path: 'resources-tmp-eEJSFC/sprites-JgnfL5/*.png' 
```
The base folder resources-tmp-* is removed after script is done. Could you show me your output?

Comment (kkostecki) [reply @here/harp-map-theme/scripts/prepareIcons.ts]: I have strong feeling that this "dir" attribute is ignored on your machine (Windows?), for sake of simplicity and flexibility I have changed implementation a little bit so it should work regardless of this attribute handling. 
Please check it out again.

Comment (okanishchev) [reply @here/harp-map-theme/scripts/prepareIcons.ts]: Archlinux actually. Seems to be the issue of ```tmp``` package.
